usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -202,6 +218,20 @@ zap_name_alloc(zap_t *zap, const char *key, matchtype_t mt)
 	}
 
 	zn->zn_hash = zap_hash(zn);
+
+	if (zap->zap_normflags) {
```

Comment:
I wonder if it would be worth skipping this if `zn_normflags == zap_normflags`(i.e. if they didn't specify `MT_PRESERVE_CASE`) (for performance reasons)?